### PR TITLE
[docs] Replace "See Also" sentences with proper blocks

### PR DESCRIPTION
**Pull request by SplitInfinity**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #63506
* #63461

**Summary**
This commit replaces some plain sentences of the form "see also <link to
function documentation page> in `_torch_docs.py` with proper "See Also"
blocks supported by RsT.

**Test Plan**
Continuous integration.

Differential Revision: [D30799704](https://our.internmc.facebook.com/intern/diff/D30799704)

### CODE REVIEW

File: torch/_torch_docs.py [hunk 1]
```diff
@@ -9566,9 +9568,10 @@ def merge_dicts(*dicts):
 Functions that return indices along a dimension, like :func:`torch.argmax` and :func:`torch.argsort`,
 are designed to work with this function. See the examples below.
 
-.. note::
-    This function is similar to NumPy's `take_along_axis`.
-    See also :func:`torch.gather`.
+.. seealso::
+        NumPy's `take_along_axis` which accepts an axis argument along which 1-D slices are taken.
```
[the remark this answers is not in the document]
Comment (SplitInfinity) [reply torch/_torch_docs.py:9572]: Should I just delete this then? Remove it from the seealso?

File: torch/_torch_docs.py [hunk 2]
```diff
@@ -9566,9 +9568,10 @@ def merge_dicts(*dicts):
...
+        :func:`torch.gather` which does not view the input as 1-D and accepts an additional argument
```
[the remark this answers is not in the document]
Comment (SplitInfinity) [reply torch/_torch_docs.py:9573]: The name of the argument is `dim`. I will add that to the sentence.

File: torch/_torch_docs.py [hunk 3]
```diff
@@ -10496,11 +10499,9 @@ def merge_dicts(*dicts):
 .. function:: where(condition) -> tuple of LongTensor
    :noindex:
 
-``torch.where(condition)`` is identical to
-``torch.nonzero(condition, as_tuple=True)``.
+.. seealso::
```
[the remark this answers is not in the document]
Comment (SplitInfinity) [reply torch/_torch_docs.py:10502]: Where should this text go? Should it replace the seealso box?

File: torch/_torch_docs.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mruberry) [reply torch/_torch_docs.py:9572]: It would actually be cool to call out the NumPy operators that are operations are based on but we don't have a template for it (yet) so yep, deleting this for now seems good.

File: torch/_torch_docs.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mruberry) [reply torch/_torch_docs.py:9573]: take_along_dim also takes a dim argument -- I would just try to summarize the biggest user-facing difference

File: torch/_torch_docs.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (mruberry) [reply torch/_torch_docs.py:10502]: Yeah I would just make it the definition of this signature without putting it in a box

File: torch/_torch_docs.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (SplitInfinity) [reply torch/_torch_docs.py:9573]: The function of the `dim` argument in the case of `torch.gather` is clear from the docs, but not so for `torch.take_along_dim`. I tried playing with the operator myself:
```
>>> sorted_idx
tensor([[[0, 0],
         [1, 1]],

        [[0, 0],
         [1, 1]]])
>>> u
tensor([[[1, 2],
         [3, 4]],

        [[5, 6],
         [7, 8]]])
>>> torch.take_along_dim(u, sorted_idx, dim=1)
tensor([[[1, 2],
         [3, 4]],

        [[5, 6],
         [7, 8]]])
>>> torch.take_along_dim(u, sorted_idx, dim=2)
tensor([[[1, 1],
         [4, 4]],

        [[5, 5],
         [8, 8]]])
>>> torch.take_along_dim(u, sorted_idx)
tensor([1, 1, 2, 2, 1, 1, 2, 2])
```
and I don't quite understand how `dim=1` and `dim=2` work. I think it's just flattening first argument to `take_along_dim` when `dim` is not specified.

**Comment by SplitInfinity** [comment]
@splitinfinity has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30799704).

File: torch/_torch_docs.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (SplitInfinity) [reply torch/_torch_docs.py:9534]: Do you see what's going on? If not, can ask the core chat tomorrow.
